client/src/App.js

```diff
@@ -41,9 +55,18 @@ const App = () => {
             /signup. Change render to component with the home page
             component if changing landing page.
         */}
-        <Route exact path="/" render={() => <Redirect to="/signup" />} />
-        <Route path="/signup" component={SignUp} />
-        <Route path="/login" component={Login} />
+        <ToastContext.Provider value={activateToast}>
+          <Route exact path="/" render={() => <Redirect to="/signup"/>}/>
+          <Route path="/signup" render={(routeProps) => <SignUp activateToast={activateToast} {...routeProps}/>}/>
```

Comment:
Could you explain what routeProps is to me? I'm guessing for this toast you'll just have it activate based on if they're successful in an action?

Reply:
routeProps is the history, match, location objects provided by the package: https://reacttraining.com/react-router/web/api/Route/route-props

For render: A bit more effiecent way I use to pass props from parent to child without using context and HOC... but now that I think about it, I need to change that, since I useContext there.

https://tylermcginnis.com/react-router-pass-props-to-components/#:~:targetText=React%20Router%20uses%20a%20declarative,app's%20location%20matches%20that%20path%20.